File: src/sp/it/pl/layout/widget/feature/PlaylistFeature.java
package sp.it.pl.layout.widget.feature;

import sp.it.pl.audio.playlist.Playlist;

/**
 * Stores list of songs to play.
 */
@Feature(
	name = "Playlist",
	description = "Stores list of songs to play",
	type = PlaylistFeature.class
)
public interface PlaylistFeature {

	/** @return playlist */
	Playlist getPlaylist();

}

File: src/player/main/sp/it/pl/layout/widget/feature/PlaylistFeature.kt
package sp.it.pl.layout.widget.feature

import sp.it.pl.audio.playlist.Playlist

/** Is bound to and manages a single playlist. */
@Feature(name = "Playlist", description = "Is bound to and manages a single playlist", type = PlaylistFeature::class)
interface PlaylistFeature {

   /** @return managed playlist */
   val playlist: Playlist

}